### PR TITLE
Eyes tracking hysteresis

### PR DESCRIPTION
**Pull request by S-Dafarra**
Improved the gaze retargeting by adding a series of hysteresis and quantizations

### CODE REVIEW

**Comment by S-Dafarra** [comment]
I think this PR is ready @kouroshD

**Comment by kouroshD** [comment]
Hi @S-Dafarra . Thank you for that, I will try to review it in this week.

**Comment by S-Dafarra** [comment]
@kouroshD friendly ping

File: modules/SRanipal_module/src/GazeRetargeting.cpp [hunk 1]
```diff
@@ -441,14 +441,41 @@ iDynTree::Vector2 GazeRetargeting::VRInterface::applyDeadzone(const iDynTree::Ve
     Eigen::Map<Eigen::Vector2d> outputMap = iDynTree::toEigen(output);
 
     double inputNorm = map.norm();
-    if (inputNorm > m_errorDeadzone)
+    bool deadzoneNotActiveAndErrorStillHigh = !m_deadzoneActive && inputNorm > m_errorDeadzone;
+    bool deadzoneActiveButErrorVeryHigh = m_deadzoneActive && inputNorm > m_errorDeadzoneActivation;
+    bool timeCheckNotActive = m_deadzoneMinActivationTimeInS <= 0.0;
+    bool timeCheckPerformedOnce = m_deadzoneActivationTime >= 0.0;
+    bool enoughTimePassed =  timeCheckNotActive || (timeCheckPerformedOnce && ((yarp::os::Time::now() - m_deadzoneActivationTime) >= m_deadzoneMinActivationTimeInS));
```
Comment (kouroshD) [inline modules/SRanipal_module/src/GazeRetargeting.cpp:448]: @S-Dafarra  can you explain a bit of strategy you took for these lines of code?

**Review by kouroshD — APPROVED** [review]

File: modules/SRanipal_module/src/GazeRetargeting.cpp (hunk 1 above)
Comment (kouroshD) [reply modules/SRanipal_module/src/GazeRetargeting.cpp:448]: What you did makes sense. But it is still hard to follow the logic here even with the meaningful variable names, at least for me. So, that's why I asked you to drop some description of the logic here.

Comment (S-Dafarra) [reply modules/SRanipal_module/src/GazeRetargeting.cpp:448]: Sorry, I think I did not understand. Can you be more specific on the logic you are referring to?

Comment (kouroshD) [reply modules/SRanipal_module/src/GazeRetargeting.cpp:448]: I am asking you to explain those lines!

**Comment by kouroshD** [comment]
@S-Dafarra before merging, can I rebase the code, or you prefer to do it?

**Comment by S-Dafarra** [comment]
> @S-Dafarra before merging, can I rebase the code, or you prefer to do it?

Yes go on, no problem :wink:
